homeassistant/components/broadlink/sensor.py

```diff
@@ -43,6 +54,30 @@
         DEVICE_CLASS_POWER,
         STATE_CLASS_MEASUREMENT,
     ),
+    "volt": (
+        "Voltage",
+        ELECTRIC_POTENTIAL_VOLT,
+        DEVICE_CLASS_VOLTAGE,
+        STATE_CLASS_MEASUREMENT,
+    ),
+    "current": (
+        "Current",
+        ELECTRIC_CURRENT_AMPERE,
+        DEVICE_CLASS_CURRENT,
+        STATE_CLASS_MEASUREMENT,
+    ),
+    "overload": (
+        "Overload",
+        ELECTRIC_CURRENT_AMPERE,
+        DEVICE_CLASS_CURRENT,
+        STATE_CLASS_MEASUREMENT,
+    ),
+    "totalconsum": (
+        "Total consumption",
+        ENERGY_KILO_WATT_HOUR,
```

Comment:
This is a summing sensor, so it requires a `last_reset` attribute if used together with `state_class`.

When only setting a `state_class` (omitting `last_reset`) is represents a current value, which this is not.

Reply:
Got it. I don't think we can get the last_reset with the python-broadlink API. We could track when the state goes to zero and get a timestamp there, but it is kindameh. Ideas? Am I missing something?

Reply:
That can work, this is how ESPHome does it:

https://github.com/home-assistant/core/pull/53592/files#diff-048f0119fa5f4366e564de0089275f935fbeecc3e0ba937d0a435c6e5cac8e41R78

Please note, this does require to restore a state on startup

Reply:
I made a working sketch. Mind having a look?

Reply:
@frenck Hi. Just to confirm. Now I just need to mark the sensor as STATE_CLASS_TOTAL_INCREASING and Home Assistant will handle the accumulation automatically?

Reply:
No need to answer, I found it [here](https://developers.home-assistant.io/blog/2021/08/16/state_class_total/). It's much better now. Thanks to whoever worked to bring this feature to the core.